doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,410 @@
+
+
+# API Description [/]
+
+## List API methods [GET]
+
+Returns an overview of available API methods with required parameters and a description.
+
++ Response 200 (text/html; charset=utf-8)
+
+
+
+# Authentication endpoint [/auth]
+
+## Authenticate [POST]
+
+The API will try to authenticate the user by the submitted credentials. In case of success a session token (JWT) will be returned, which can be used for further requests to the API.
+
++ Request (application/x-www-form-urlencoded)
+    + Attributes
+        + username: the_username (string, required)
+        + password: aSecretPassword (string, required)
+        
+    + Schema
+    
+            {
+                "type": "object",
+                "properties": {
+                    "username": {
+                        "description": "Name of the user that wants to authenticate",
+                        "type": "string"
+                    },
+                    "password": {
+                        "description": "Password for the username in question",
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "username",
+                    "password"
+                ],
+                "additionalProperties": false
+            }
+ 
++ Response 200 (application/json)
+    + Body
+
+            {
+                "status": 200,
+                "token": "eyJ...Q2tY="
+            }
+            
+    + Schema
+    
+            {
+                "type": "object",
+                "properties": {
+                    "status": {
+                        "description": "HTTP status code",
+                        "type": "integer",
+                        "minimum": 0
+                    },
+                    "token": {
+                        "description": "Session token to be used as authentication for further requests.",
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "status",
+                    "token"
+                ],
+                "additionalProperties": false
+            }
+
++ Response 401 (application/json)
+    + Body
+    
+            {
+                "status": 401,
+                "error": "Unauthorized",
+                "message": "You have to supply an username and a password.",
+                "loca": "AUTH_NEED_USERNAME_PASSWORD"
+            }
+            
+    + Schema
+
+            {
+                "type": "object",
+                "properties": {
+                    "status": {
+                        "description": "HTTP status code",
+                        "type": "integer",
+                        "minimum": 0
+                    },
+                    "error": {
+                        "description": "Textual description of HTTP status code",
```

Comment:
Actually we could ommit the textual status code description completely, as every browser does translate codes anyway. I just thought it could be useful when using command line tools like cURL.

The localized status is always something more detailed, like a hint about missing credentials or wrong format or something like that. When the API is final, I would create a document containing a mapping for i18n key and meaning.